configs/components/pxp-agent.rb

```diff
@@ -93,7 +93,51 @@
   when "aix"
     pkg.install_service "resources/aix/pxp-agent.service", nil, "pxp-agent"
   when "windows"
-    puts "Service files not enabled on windows"
+    service_hash = [{
+      :directory_ref => settings[:bindir_id],
+      :id => "service_nssm",
+      :guid => "52B1CD57-95A2-4CA4-AB8E-9DDD6DE8FC66",
+      :file => {
+        :id => "NSSM",
+        :source => settings[:nssm],
+      },
+      :serviceinstall => {
+        :id => "PXPServiceInstaller",
+        :description => "Puppet Execution Protocol (PXP) Service",
+        :displayname => "Puppet PXP Agent",
+        :name => "pxp-agent",
+        :start => "demand",
+      },
+      # Various registry keys documented at https://nssm.cc/usage
+      :registries => {
+        :root => "HKLM",
+        :key => "SYSTEM\\CurrentControlSet\\Services\\pxp-agent\\Parameters",
+        :values => [
+          { :name => "AppDirectory", :value => "[INSTALLDIR]\\bin", :type => "expandable" },
```

Comment:
Sorry for snapping. It's good to double check and make sure things are consistent.